pkg/controller/database_snapshot.go

```
@@ -0,0 +1,309 @@
+					c.create(dbSnapshot)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				dbSnapshot := obj.(*tapi.DatabaseSnapshot)
+				c.delete(dbSnapshot)
+			},
+		},
+	)
+	cacheController.Run(wait.NeverStop)
+}
+
+const (
+	durationCheckSnapshotJob = time.Minute * 30
+)
+
+func (c *DatabaseSnapshotController) create(dbSnapshot *tapi.DatabaseSnapshot) {
+	// Validate DatabaseSnapshot spec
+	if err := c.snapshoter.Validate(dbSnapshot); err != nil {
+		c.eventRecorder.PushEvent(kapi.EventTypeWarning, eventer.EventReasonInvalid, err.Error(), dbSnapshot)
+		log.Errorln(err)
+		return
+	}
+
+	runtimeObj, err := c.snapshoter.GetDatabaseRuntimeObject(dbSnapshot)
+	if err != nil {
+		c.eventRecorder.PushEvent(kapi.EventTypeWarning, eventer.EventReasonFailedToGet, err.Error(), dbSnapshot)
+		log.Errorln(err)
+		return
+	}
+
+	c.eventRecorder.PushEvent(
+		kapi.EventTypeNormal, eventer.EventReasonStarting, "Starting backup",
+		runtimeObj, dbSnapshot,
+	)
+
+	job, err := c.snapshoter.GetSnapshotObjects(dbSnapshot)
+	if err != nil {
+		message := fmt.Sprintf(`Failed to take snapshot. Reason: %v`, err)
+		c.eventRecorder.PushEvent(
+			kapi.EventTypeWarning, eventer.EventReasonSnapshotFailed, message,
+			runtimeObj, dbSnapshot,
+		)
+		log.Errorln(err)
+		return
+	}
+
+	if _, err := c.client.Batch().Jobs(dbSnapshot.Namespace).Create(job); err != nil {
+		message := fmt.Sprintf(`Failed to take snapshot. Reason: %v`, err)
+		c.eventRecorder.PushEvent(
+			kapi.EventTypeWarning, eventer.EventReasonSnapshotFailed, message,
+			runtimeObj, dbSnapshot,
+		)
+		log.Errorln(err)
+		return
+	}
+
+	c.eventRecorder.PushEvent(
+		kapi.EventTypeNormal, eventer.EventReasonSuccessfulSnapshot, "Successfully completed snapshot",
+		runtimeObj, dbSnapshot,
+	)
+
+	go c.checkDatabaseSnapshotJob(dbSnapshot, job.Name, durationCheckSnapshotJob)
+}
+
+func (c *DatabaseSnapshotController) delete(dbSnapshot *tapi.DatabaseSnapshot) {
+	runtimeObj, _ := c.snapshoter.GetDatabaseRuntimeObject(dbSnapshot)
```

Why ignore this error?